### PR TITLE
refactor: svg icon & functions (auth)

### PR DESCRIPTION
**Pull request by MKMonkeyCat**
# Description
- Refactored SVG icon component
- Refactoring functions related to Oauth

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changesh my changes

### CODE REVIEW

**Comment by MKMonkeyCat** [comment]
done

File: src/utils/Object.ts [hunk 1]
```diff
@@ -37,3 +45,20 @@ export const deepAssign = (
   });
   return result;
 };
+
+/** dict omit
+ * Returns a new object with properties from the original object omitted.
+ * for typescript Omit type
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const DictOmit = <T extends Record<keyType, any>, K extends keyType>(
```
Comment (SiongSng) [inline src/utils/Object.ts:54]: 這個函式的用途是?
沒看到任何地方有用到

**Review by SiongSng — CHANGES_REQUESTED** [review]

File: src/utils/Object.ts (hunk 1 above)
Comment (qmkcat) [reply src/utils/Object.ts:54]: 基於 typescript Omit，函數實例

Comment (qmkcat) [reply src/utils/Object.ts:54]:
```ts
const t = {
  a: 'a',
  b: 'b',
};
const a = DictOmit(t, 'a');
// a => {
//   b: 'b'
// }
```

Comment (SiongSng) [reply src/utils/Object.ts:54]: ok :+1:  不錯的功能
但我看好像沒其他地方調用 (call) 到這個 function，可以先刪除嗎？有用到再加回來

Comment (SiongSng) [reply src/utils/Object.ts:54]: @a3510377 所以你覺得如何? .w.

Comment (qmkcat) [reply src/utils/Object.ts:54]: 痾就留著八原本有用到所以寫出來的以後應該還會用到

**Review by SiongSng — APPROVED** [review]
LGTM

**Pull request merged**